### PR TITLE
Change from permanent redirect to temporary

### PR DESCRIPTION
**Pull request by wshafer**
Provide a narrative description of what you are trying to accomplish:

- [x] Are you fixing a bug?

Updated the unauthorizedResponse to send out a 302 instead of a 301, as a 301 can cause issues with the browser on subsequent requests, since 301 redirects are cached.

### CODE REVIEW

File: src/PhpSession.php [hunk 1]
```diff
@@ -103,7 +103,7 @@ public function unauthorizedResponse(ServerRequestInterface $request) : Response
                 'Location',
                 $this->config['redirect']
             )
-            ->withStatus(301);
+            ->withStatus(302);
```
Comment (michalbundyra) [inline src/PhpSession.php:106]: Hm... or maybe it should be configurable? I can see we have `$this->config` here so users can chose what they want to use.

Comment (geerteltink) [reply src/PhpSession.php:106]: @wshafer is correct. As the 301 code is a permanent redirect, browser may (and will) cache it. I've experienced this as well.

**Comment by weierophinney** [comment]
@wshafer You'll need to update test expectations as well; see [this test failure](https://travis-ci.org/zendframework/zend-expressive-authentication-session/jobs/355536888#L659) for why.

File: src/PhpSession.php (hunk 1 above)
Comment (weierophinney) [reply src/PhpSession.php:106]: @webimpress Definitely needs to be a 302; it's **not** permanent, as after authentication, they should be able to proceed to the originally requested resource.

**Review by geerteltink — APPROVED** [review]

**Comment by weierophinney** [comment]
Thanks, @wshafer!

**Pull request merged**